cloud-common/src/test/java/com/sequenceiq/cloudbreak/util/FixedSizePreloadCacheTest.java

```diff
@@ -16,14 +13,4 @@ public void testPopWhenOnTheFlyGenerationNeeded() {
 
         assertEquals(Thread.currentThread().getName(), actual);
     }
-
-    @Test
```

Comment:
Would be better to put @Ignore instead of delete this method, wouldn't be?

Reply:
I saw no way to make this specific testcase stable, so I removed it.

Reply:
One way to make it more deterministic is the following change: https://github.com/adoroszlai/cloudbreak/commit/b5b72fa6a60d9cee2a9a9bbef6ab223622845c9f

The original unit test failure can be reproduced by adding `Thread.sleep` before the `synchronized` block in: https://github.com/hortonworks/cloudbreak/blob/8b865421af1489ea6816ec4d486eaf7de90e005b/cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/FixedSizePreloadCache.java#L35-L41

Reply:
Cool. I was also playing around with the Thread.sleep value, but the await is much nicer. I'll apply your version to this branch.